pkg/runtime/interfaces/cluster_pools.go

```diff
@@ -0,0 +1,17 @@
+package interfaces
+
+//go:generate mockery -name ClusterPoolAssignmentConfiguration -output=mocks -case=underscore
+
+type ClusterPoolAssignment struct {
+	Pool string `json:"pool"`
+}
+
+type ClusterPoolAssignments = map[DomainName]ClusterPoolAssignment
```

Comment:
Are these strictly going to be defined only per domain, what if we needed it also at project-domain level or is that not something we are considering for this

Reply:
Already implemented as a matchable attribute https://github.com/flyteorg/flyteadmin/pull/376